app/policies/request_policy.rb

```diff
@@ -6,19 +6,27 @@ class Scope < ApplicationPolicy::Scope
 
     def resolve
       return scope.all unless user.rbac_enabled?
-      klass = scope == Request ? scope : scope.model
-
-      case Insights::API::Common::Request.current.headers[Insights::API::Common::Request::PERSONA_KEY]
-      when PERSONA_ADMIN
-        raise Exceptions::NotAuthorizedError, "No permission to access the complete list of requests" unless admin?(klass)
-        scope == Request ? scope.where(:parent_id => nil) : scope
-      when PERSONA_APPROVER
-        raise Exceptions::NotAuthorizedError, "No permission to access requests assigned to approvers" unless approver?(klass)
-        approver_visible_requests(scope)
-      when PERSONA_REQUESTER, nil
-        scope == Request ? scope.by_owner.where(:parent_id => nil) : scope
+
+      if user.params[:request_id]
+        req = Request.find(user.params[:request_id])
+        raise Exceptions::NotAuthorizedError, "Read access not authorized for request #{req.id}" unless resource_check('read', req)
+        req.children
+      # GraphQL comes in with model_class.all, whose scope is respond to model call
```

Comment:
@hsong-rh Can't we fix this here
https://github.com/RedHatInsights/approval-api/blob/830f34c87db22b513b70a301f1907e4d80b29270/app/controllers/api/v1x0/graphql_controller.rb#L20

Always call model

Reply:
@mkanoor we use model or scope to distinguish calling from regular controller or GraphQL. The result is different.

Reply:
@mkanoor Per our discussion online, will refactor this logic after graphql is fixed in insights-api-common.